internal/mocks/mock_error_iterator.go

```diff
@@ -0,0 +1,50 @@
+package mocks
```

Comment:
I'm not convinced by this package name. This isn't a mock, this is an implementation of a tupleiterator that does a specific thing. The fact that it should be only used in tests should probably be mentioned in the godoc of `NewErrorIterator`

Reply:
Are there mocks that are not used for tests?